api/config/v1alpha1/envoygateway_types.go

```diff
@@ -52,6 +54,12 @@ type Gateway struct {
 	//
 	// +optional
 	ControllerName string `json:"controllerName,omitempty"`
+
+	// ProxyImage is the optional proxy image to use in
+	// the Infra IR to deploy EnvoyProxy. If unspecified, the default proxy
+	// image will be used.
+	// +optional
```

Comment:
- Since this is a user-facing API, I think it's better to s/in the Infra IR to deploy EnvoyProxy/for the managed Envoy proxies/
- IMO this field does not belong in the `Gateway` struct` and I'm unsure if we want to expose the Envoy version to users. Making the version configurable can create a support nightmare of neverending EG<>Envoy combinations. IMO we release EG with the latest Envoy release and users upgrade EG if they need a newer version of Envoy. Thoughts @arkodg @LukeShu @skriss @youngnick @AliceProxy?
- Add an empty line before `+optional`.

Reply:
+1 to delaying this API change until users really need it. Post v0.3.0 an exercise we will need to do is
* what level of customizations do we need to provide for
  * xds
  * EG platform/k8s deployment/service
  * Envoy platform/k8s deployment/service
* how do we expose these customizations

Reply:
> I think it should be updated in the configuration...

If we expose the Envoy image here, then users can change the image. I don't think we should do this due to the concerns I stated in https://github.com/envoyproxy/gateway/pull/712#discussion_r1016856370.

Reply:
Yes, I understand that @danehans. But for resolving https://github.com/envoyproxy/gateway/issues/672, how do we do it without exposing configuration? By just changing the default image const? We keep the const to the envoy-dev:latest in main. Everytime we release, we update the const to the appropriate envoy release version.

Reply:
> We keep the const to the envoy-dev:latest in main. Everytime we release, we update the const to the appropriate envoy release version.

Yes.

Reply:
Okay, if so, this is just a easy-update. I have updated. PTAL @danehans @arkodg.